packages/@lwc/engine-core/src/framework/base-lightning-element.ts

```diff
@@ -230,7 +220,15 @@ export const LightningElement: LightningElementConstructor = function (
     associateVM(component, vm);
     associateVM(elm, vm);
 
-    initializeMode(vm);
+    if (!features.ENABLE_LIGHT_DOM_COMPONENTS) {
+        const { ctor, renderMode } = def;
+        assert.isTrue(
+            renderMode !== 'light',
+            `${
+                ctor.name || 'Anonymous class'
```

Comment:
anything that you need from ctor, should be extracted into def rather than accessing it on the fly, but I get this is dev-time assertion.

Reply:
The name is already part of the `ComponentDef`.